modules/katello_activation_key.py

```diff
@@ -256,39 +158,57 @@ def main():
     if has_import_error:
         module.fail_json(msg=import_error_msg)
 
-    server_url = module.params['server_url']
-    username = module.params['username']
-    password = module.params['password']
-    verify_ssl = module.params['verify_ssl']
-    name = module.params['name']
-    organization = module.params['organization']
-    lifecycle_environment = module.params['lifecycle_environment']
-    content_view = module.params['content_view']
-    subscriptions = module.params['subscriptions']
-    auto_attach = module.params['auto_attach']
-    new_name = module.params['new_name']
-    state = module.params['state']
+    entity_dict = dict(
+        [(k, v) for (k, v) in module.params.items() if v is not None])
 
-    server = ServerConfig(
-        url=server_url,
-        auth=(username, password),
-        verify=verify_ssl
-    )
-    ng = NailGun(server, entities, module)
+    server_url = entity_dict.pop('server_url')
+    username = entity_dict.pop('username')
+    password = entity_dict.pop('password')
+    verify_ssl = entity_dict.pop('verify_ssl')
+    state = entity_dict.pop('state')
 
-    # Lets make an connection to the server with username and password
     try:
-        org = entities.Organization(server)
-        org.search()
+        create_server(server_url, (username, password), verify_ssl)
     except Exception as e:
         module.fail_json(msg="Failed to connect to Foreman server: %s " % e)
 
+    ping_server(module)
+
+    entity_dict['organization'] = find_organization(module, name=entity_dict['organization'])
+    if entity_dict['lifecycle_environment']:
```

Comment:
So, LE/CV aren't required, but if you don't provide them, this check will fail with a `KeyError`. I think you want to check the key exists first (this isn't Ruby).

```py
if 'lifecycle_environment' in entity_dict and entity_dict['lifecycle_environment']:
  ...
```

Or, another solution, set these keys to be None by default.